MapboxCoreNavigation/MBNavigator.swift

```diff
@@ -7,10 +7,10 @@ extension MBFixLocation {
     convenience init(_ location: CLLocation) {
         self.init(coordinate: location.coordinate,
                   time: location.timestamp,
-                  speed: location.speed as NSNumber,
-                  bearing: location.course as NSNumber,
+                  speed: location.speed >= 0 ? location.speed as NSNumber : nil,
+                  bearing: location.course >= 0 ? location.course as NSNumber : nil,
                   altitude: location.altitude as NSNumber,
```

Comment:
altitude is nullable but supports negative values to indicate below sea level so this conversion is correct

Reply:
yep that was my thoughts exactly. we want to have correct altitudes when passing through death valley or doing u-boot routing ;)